runtime/oti/VMHelpers.hpp

```diff
@@ -2048,18 +2049,56 @@ class VM_VMHelpers
 #endif /* JAVA_SPEC_VERSION > 11 */
 	}
 
-	static VMINLINE UDATA
-	walkContinuationStackFramesWrapper(J9VMThread *vmThread, j9object_t continuationObject, J9StackWalkState *walkState)
-	{
-		UDATA rc = J9_STACKWALK_RC_NONE;
 #if JAVA_SPEC_VERSION >= 19
-		J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(vmThread, continuationObject);
-		rc = vmThread->javaVM->internalVMFunctions->walkContinuationStackFrames(vmThread, continuation, walkState);
-#endif /* JAVA_SPEC_VERSION >= 19 */
-		return rc;
+/*
+ *	If low tagging failed (or not even attempted) due to either
+ *
+ *   a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
+ *   another GC thread winning to scan, again don't do anything, and let the winning thread do the work, instead
+ */
+	static VMINLINE bool
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = &continuation->state;
+		uintptr_t oldContinuationState = *localAddr;
+
+		if (0 == oldContinuationState) {
```

Comment:
could use helpers to interpret the state.
either combine 2 existing ones (carrier is null and not concurrently scanning) or even create a new one 

but there would still be the atomic op left dealing with raw state and and raw flag const
then, even the atomic could be put into a helper (atomicallySetConcurentlyScannedIntoState(localAddr, oldState)

not sure, perhaps it would be going too far. i'll leave it up to you

Reply:
After a bit of thinking:
- we should introduce #define J9_GC_CONTINUATION_STATE_INITIAL  0 (and add STATE in J9_GC_CONTINUATION_CONCURRENTSCANNING)
- we could even skip the check altogether (hence no need for new helper). That check will be true most of the time, so we may as well go straight to atomic, without even reading old value. 

The whole method collapses to just a single line:

return J9_GC_CONTINUATION_STATE_INITIAL == VM_AtomicSupport::lockCompareExchange(&continuation->state, J9_GC_CONTINUATION_STATE_INITIAL, J9_GC_CONTINUATION_STATE_CONCURRENT_SCANNING);